jvb/src/main/kotlin/org/jitsi/videobridge/octo/config/OctoConfig.kt

```diff
@@ -16,151 +16,75 @@
-                    retrievedAs<ConfigObject>() convertedBy {
-                        val cfg = it.toConfig()
-                        if (cfg.hasPath("BIND_ADDRESS") && cfg.hasPath("BIND_PORT")) {
-                            val bindAddress = cfg.getString("BIND_ADDRESS")
-                            val bindPort = cfg.getInt("BIND_PORT")
-                            bindAddress != null && (bindPort.isUnprivilegedPort())
-                        } else {
-                            false
-                        }
-                    }
-                },
-                newConfigAttributes {
-                    name("videobridge.octo.enabled")
-                    readOnce()
-                }
-            )
-
-            private val enabledProp = EnabledProperty()
-
-            @JvmStatic
-            fun enabled() = enabledProp.value
-
-            class RegionProperty : LegacyFallbackConfigProperty<String>(
-                String::class,
-                "org.jitsi.videobridge.REGION",
-                "videobridge.octo.region",
-                readOnce = true
-            )
-
-            private val regionProp = RegionProperty()
-
-            @JvmStatic
-            fun region(): String? {
-                return try {
-                    regionProp.value
-                } catch (t: Throwable) {
-                    null
-                }
+    val recvQueueSize: Int by config("videobridge.octo.recv-queue-size".from(JitsiConfig.newConfig))
+
+    val sendQueueSize: Int by config("videobridge.octo.send-queue-size".from(JitsiConfig.newConfig))
+
+    // We grab these two properties from the legacy config separately here
+    // because we use them to infer a legacy value of 'enabled' (which was
+    // based on the presence of these properties) and as potential values
+    // in each of the individual bindAddress and bindPort properties.
+    private val legacyBindAddress: String? by optionalconfig(
+        "org.jitsi.videobridge.octo.BIND_ADDRESS".from(JitsiConfig.legacyConfig))
+    private val legacyBindPort: Int? by optionalconfig(
+        "org.jitsi.videobridge.octo.BIND_PORT".from(JitsiConfig.legacyConfig))
+
+    val enabled: Boolean by config {
+        // The legacy config file doesn't have an 'enabled' property,
+        // instead it was based on the values of the parameters.  Here,
+        // we simulate a legacy 'enabled' value based on the results
+        // of validating the other properties in the legacy config
+        // file.  If neither property is present, we consider the field
+        // "not found" in the legacy config.
+        "Legacy Octo relay enabled" {
+            if (legacyBindAddress == null && legacyBindPort == null) {
+                throw ConfigException.UnableToRetrieve.NotFound("not found")
             }
+            legacyBindAddress != null && legacyBindPort?.isUnprivilegedPort() == true
+        }
+        "videobridge.octo.enabled".from(JitsiConfig.newConfig)
```

Comment:
Just curious if I understand this correctly: if the legacy bind address or port are unset, the first block will throw `NotFound`, so we'll retrieve the new property?

Reply:
Yeah...we didn't have an 'enabled' property before, the code itself checked that both bind address and port were set, so that's what's being replicated here.  (also, if the port is set but isn't an 'unprivileged port', it'll fall through)